backend/web/src/frontend/src/views/header/component/TagNavigationBody.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+  <v-card :flat="true">
+    <v-list :key="category.id" class="grow" v-for="category in categories">
+      <CategoryGroup :category="category" />
+    </v-list>
+  </v-card>
+</template>
+
+<script>
+import { FETCH_CATEGORIES } from '@/store/share/actionTypes.js';
+import { mapActions, mapGetters } from 'vuex';
+import { SEARCHED_CATEGORIES, TOTAL_CATEGORIES } from '@/store/share/getterTypes.js';
+import CategoryGroup from '@/views/header/component/CategoryGroup.vue';
+
+export default {
+  name: 'TagNavigationBody',
+  components: {
+    CategoryGroup,
+  },
+  props: {
+    searchKeyword: {
+      type: String,
+    },
+  },
+  computed: {
+    ...mapGetters([TOTAL_CATEGORIES, SEARCHED_CATEGORIES]),
+    categories() {
+      if (this.searchKeyword === '') {
+        return this[TOTAL_CATEGORIES];
+      }
+      return this[SEARCHED_CATEGORIES](this.searchKeyword);
+    },
```

Comment:
아래처럼 사용한다면, TOTAL_CATEGORIES와 SEARCHED_CATEGORIES를 구분해서 getter를 따로 만들 필요가 없을 것이고, computed에서 드러나있는 세부 구현 내용을 category의 getter하나안에 캡슐화 할수 있을 것 같아요.

```suggestion
// @/store/modules/category.js
// ...

const getters = {
  [SEARCHED_CATEGORIES](state) => (searchKeyword) => {
    if (!searchKeyword) {
        return state.categories;
    }
    const totalCategories = JSON.parse(JSON.stringify(state.categories));
    totalCategories.forEach((category) => {
            category.tags = category.tags.filter(
                (tag) => tag.name.includes(searchKeyword));
    });

    return totalCategories;
  },
};

```

Reply:
이걸 고민안한게 아닌데 이렇게 getter를 사용해도될지.......ㅎㅎㅎ

Reply:
@taejinseok 
이부분에 대해서 궁금한게 있는데 
getter 라는 naming만보면 그냥 값을 가져오는느낌인데 조건에 따라 return 을 다르게 가져오는 형태가 괜찮은지 궁금해요!
naming 을 잘하는건 아니지만..
searchKeyword 에 따라 categories값이 return 되니까 
제 의견은 getCategoriesBySearchKeyword 처럼 안의 로직을 어느정도 예상할 수 있는 naming이 좋지 않을까 하는생각입니다.
어떻게 생각하시는지 궁금해요!

Reply:
@taejinseok
이부분에 대해서 궁금한게 있는데
getter 라는 naming만보면 그냥 값을 가져오는느낌인데 조건에 따라 return 을 다르게 가져오는 형태가 괜찮은지 궁금해요!
naming 을 잘하는건 아니지만..
searchKeyword 에 따라 categories값이 return 되니까
제 의견은 getCategoriesBySearchKeyword 처럼 안의 로직을 어느정도 예상할 수 있는 naming이 좋지 않을까 하는생각입니다.
어떻게 생각하시는지 궁금해요!